src/containers/my-resources/lessons-container/LessonsContainer.tsx

```diff
@@ -0,0 +1,216 @@
+import { ChangeEvent, useCallback, useRef, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { useNavigate } from 'react-router-dom'
+import SearchIcon from '@mui/icons-material/Search'
+import Box from '@mui/material/Box'
+
+import AppButton from '~/components/app-button/AppButton'
+import AppDrawer from '~/components/app-drawer/AppDrawer'
+import AppPagination from '~/components/app-pagination/AppPagination'
+import EnhancedTable from '~/components/enhanced-table/EnhancedTable'
+import InputWithIcon from '~/components/input-with-icon/InputWithIcon'
+import Loader from '~/components/loader/Loader'
+import {
+  columns,
+  initialSort,
+  itemsLoadLimit,
+  removeColumnRules
+} from '~/containers/my-resources/lessons-container/LessonsContainer.constants'
+import usePagination from '~/hooks/table/use-pagination'
+import useSort from '~/hooks/table/use-sort'
+import useAxios from '~/hooks/use-axios'
+import useBreakpoints from '~/hooks/use-breakpoints'
+import useConfirm from '~/hooks/use-confirm'
+import { useDebounce } from '~/hooks/use-debounce'
+import { useDrawer } from '~/hooks/use-drawer'
+import { ResourceService } from '~/services/resource-service'
+
+import { AxiosResponse } from 'axios'
+import { defaultResponses, snackbarVariants } from '~/constants'
+import { styles } from '~/containers/my-resources/lessons-container/LessonsContainer.styles'
+import { useSnackBarContext } from '~/context/snackbar-context'
+import { authRoutes } from '~/router/constants/authRoutes'
+import { ErrorResponse, ItemsWithCount, Lesson, Sort } from '~/types'
+import {
+  ajustColumns,
+  createUrlPath,
+  getScreenBasedLimit
+} from '~/utils/helper-functions'
+
+interface LessonsParams {
+  title: string
+  skip: number
+  limit: number
+  sort: Sort
+}
```

Comment:
You can extends from this interface.

![image](https://github.com/ita-social-projects/SpaceToStudy-Client/assets/83120263/6a564003-9ecf-48b5-bb35-c37cccff9ca6)